pymc3/gp/gp.py

```diff
@@ -136,8 +136,8 @@ def sample_gp(trace, gp, X_values, samples=None, obs_noise=True, model=None, ran
     # Posterior covariance
     S_post = S_zz - tt.dot(tt.dot(S_xz.T, S_inv), S_xz)
 
-    gp_post = MvNormal.dist(m_post, S_post, shape=Z.shape[0])
-    
+    gp_post = MvNormal.dist(m_post, S_post + jitter * tt.eye(S_post.shape[0]), shape=Z.shape[0])
```

Comment:
I see your point RE "jitter", I was thinking of it as adding teeny bit of white noise to the sample predictions.  Not sure about cov_const, since in a model one could have a covariance function that is

    cov_func = ExpQuad(1, l) + c

where `c` is a constant.  Might be confusing?  What about `num_stability_const`?  (long but descriptive?).

What about not having a parameter, but making this value a small multiple of the trace of `S_post`?

    n = S_post.shape[0]  
    1e-8 * tt.nlinalg.trace(S_post)/n * tt.eye(n)

Hopefully the `1e-8` value could be calibrated such that the correction is large enough to make the Cholesky numerically stable when the `S_post` is pos. def., but not high enough to fix `S_post` if it actually is not pos. def.